src/mkdocstrings_handlers/python/handler.py

```diff
@@ -235,11 +236,12 @@ def collect(self, identifier: str, config: Mapping[str, Any]) -> CollectorItem:
                 modules_collection=self._modules_collection,
                 lines_collection=self._lines_collection,
             )
-            try:
+            try:  # noqa: WPS229 we expect one type of exception, and want to fail on the first one
+                for pre_loaded_module in config.get("preload_external_modules", []):
+                    loader.load_module(pre_loaded_module)
```

Comment:
That's what I wanted to avoid: loading every time we collect a new thing. That's what I meant by "as early as possible" in our discussion: I meant to accept the preload option as a global option, available when instantiating the handler, so that we can preload only once. Maybe you tried that but it didn't work? I'll check on my side how feasible it is.

Reply:
I went on that path since I was not sure if the config in the constructor is always complete. 

If it is - then I can move this to the constructor. In the meantime, I have added a check if the module was already loaded to avoid double work.

Reply:
I assume you will know if the config in the constructor is good enough. Let me know if you want me to move this to the constructor.

Currently it's lazy computed which is almost as efficient.

Do you have any other comments or things you want me to change?